hedera-node/src/main/java/com/hedera/services/state/submerkle/EvmFnResult.java

```diff
@@ -80,6 +78,7 @@ public class EvmFnResult implements SelfSerializable {
 	private EntityId contractId;
 	private List<EntityId> createdContractIds = Collections.emptyList();
 	private List<EvmLog> logs = Collections.emptyList();
+	// stateChanges are only kept for migration purposes, they are not saved with the txn record anymore
 	private Map<Address, Map<Bytes, Pair<Bytes, Bytes>>> stateChanges = Collections.emptyMap();
```

Comment:
We should delete this field now; since no production state includes any `stateChanges`, it is in fact safe to delete [this line](https://github.com/hashgraph/hedera-services/blob/03401-record-stream-sidecar-support/hedera-node/src/main/java/com/hedera/services/state/submerkle/EvmFnResult.java#L223) in `deserialize()`.

Reply:
Deleted now. However, some awkward code for reading state changes is left in order to read correctly the serialized `EvmFnResult`s from previous versions for the serdes tests, which were serialized with actual state changes... Is the current state OK?

Reply:
I think this is good...in 0.29 we can clean up this dead code.